pandas/core/arrays/categorical.py

```diff
@@ -1352,8 +1353,7 @@ def memory_usage(self, deep=False):
         """
         return self._codes.nbytes + self.dtype.categories.memory_usage(deep=deep)
 
-    @Substitution(klass="Categorical")
-    @Appender(_shared_docs["searchsorted"])
+    @doc(IndexOpsMixin.searchsorted, klass="Categorical")
```

Comment:
I agree with this comment here - this seems strange to use `IndexOpsMixin` since it isn't part of this classes hierarchy. Any reason not to address this?